### PR TITLE
Add is_equivalent() convenience method for Quantity and test

### PR DESCRIPTION
**Pull request by parejkoj**
`astropy.unit` objects can test whether they are equivalent to each other. It's useful to be able to test whether two quantities are in equivalent units, or whether a quantity is equivalent to a unit (or unit representation). This PR adds such a method on Quantity, with a few tests. I didn't see find the tests for `unit.is_equivalent()`, but I'm leveraging the assumption that it behaves itself, and not writing a full set of tests for the Quantity version.

I wasn't sure how to document the new method so that it's not just duplicating the full docstring of `unit.is_equivalent()`: I'd rather just change the bits that are different (e.g. `other` can be a Quantity, in addition to all other types accepted by the `unit` method), and for everything else tell the user to refer to the `unit.is_equivalent()` method.

### CODE REVIEW

File: astropy/units/tests/test_quantity.py [hunk 1]
```diff
@@ -667,6 +667,13 @@ def test_quantity_comparison(self):
 
         assert u.Quantity(1200, unit=u.meter) != u.Quantity(1, unit=u.kilometer)
 
+    def test_quantity_is_equivalent(self):
```
Comment (pllim) [inline astropy/units/tests/test_quantity.py]: What about composite units? Say, `erg / s / cm^2 / Hz^2`? Is that considered equivalent to `Jy`?

Comment (parejkoj) [inline astropy/units/tests/test_quantity.py]: I have no idea, I just hand the comparison off to the unit comparison code. Is a composite unit a `Unit` or a `Quantity`?

File: astropy/units/quantity.py [hunk 2]
```diff
@@ -768,6 +768,23 @@ def to(self, unit, equivalencies=[]):
...
+
+        Convenience method for `quantity.unit.is_equivalent()`
+
+        Parameters
+        ----------
+        other : Quantity or Unit or string or tuple
+            The unit to convert to. If a tuple of units is specified, this
+            method returns true if the unit matches any of those in the tuple.
+        """
+        if isinstance(other, Quantity):
```
Comment (adrn) [inline astropy/units/quantity.py]: To be even more agnostic, how about:
```python
if hasattr(other, 'unit'):
    return self.unit.is_equivalent(other.unit)
else:
    return self.unit.is_equivalent(other)
```

File: astropy/units/quantity.py [hunk 3]
```diff
@@ -768,6 +768,23 @@ def to(self, unit, equivalencies=[]):
...
+            return self.unit.is_equivalent(other.unit)
```
Comment (adrn) [inline astropy/units/quantity.py]: BTW: you still need to pass through the equivalencies.

File: astropy/units/tests/test_quantity.py [hunk 4]
```diff
@@ -667,6 +667,13 @@ def test_quantity_comparison(self):
...
+        assert u.Quantity(10, unit='m').is_equivalent(u.Quantity(1, unit='cm'))
+        assert u.Quantity(10, unit='m').is_equivalent(u.pc)
+        assert u.Quantity(10, unit='kg').is_equivalent('g')
+        assert not u.Quantity(10, unit='m').is_equivalent(u.Quantity(1, unit='mag'))
+        assert not u.Quantity(10, unit='m').is_equivalent(u.arcsec)
+
```
Comment (adrn) [inline astropy/units/tests/test_quantity.py]: I would add a test passing through a unit with equivalencies, e.g., using the `u.spectral()` equivalency.

**Comment by mhvk** [comment]
Sorry to come to this late, but I'm a bit lost as to why we want this new method. It is not much more work to type `.unit.is_equivalent` and the advantage of the latter is that equivalency of units has a clear meaning, while equivalency of quantities to me is very unclear. (And why just the unit? Isn't it just as important that, e.g., shapes can be broadcasted?)

**Comment by parejkoj** [comment]
I think of unit equivalency as different from shape equivalency.

**Comment by mhvk** [comment]
Could you please give a use case? I do not think we should lightly add new methods to something stable in astropy (which `Quantity` is).

**Comment by adrn** [comment]
I was initially neutral on this, but on reflection, I tend to agree with @mhvk

**Comment by parejkoj** [comment]
The use-case I have is validating measurements (Quantities) of defined metrics (that have specified units). I need to check that each `Quantity` that comes in has `units` that are equivalent to the `units` of the definition.

**Comment by mhvk** [comment]
What about using `quantity_input`? (http://docs.astropy.org/en/latest/units/quantity.html#functions-accepting-quantities)

Though I should note that, after having completely converted to `Quantity`, I tend to just duck-type: the code will raise a `UnitsError` anyway if things are not OK. Since `is_equivalent` is also quite an expensive operation [1], I'd generally advocate using something like:
```
try:
   <your-quantity-operation>
except u.UnitsError:
    raise MoreInformativeErrorMessage
```

[1] For small arrays, about a third of the cost of doing the operation:
```
In [2]: q = np.arange(10)*u.m

In [3]: q2 = 5 * u.m

In [4]: %timeit q+q2
10000 loops, best of 3: 42 µs per loop

In [5]: %timeit q.unit.is_equivalent(q2.unit)
100000 loops, best of 3: 13.4 µs per loop
```

**Comment by parejkoj** [comment]
No, I can't duck type in this instance: this is essentially testing that the input data conforms to the API (it may get stuffed into a database, or written to a file immediately afterward).

**Comment by parejkoj** [comment]
`@quantity_input` doesn't help in this instance, since the required units are defined outside the python code (in a YAML file, in this case).

**Comment by MSeifert04** [comment]
@parejkoj But given that the actual function only is 4 lines long (it could be one line when you use `getattr(other, 'unit', other)`  it shouldn't be much of a problem to implement it wherever you need it, right?

**Pull request closed without merging**